dio/lib/src/entry/dio_for_native.dart

```diff
@@ -120,18 +121,40 @@ class DioForNative with DioMixin implements Dio {
       file = File(savePath.toString());
     }
 
+    // For the file resumption option.
+    // Its 0 for a new file.
+    int fileLength = 0;
+
+     if (resumeDownload) {
+      bool fileExists = await file.exists();
+
+      if (!fileExists) {
+        //For a resume uploads operation, the file must exist in the given directory.
+        //It it doesnt exist, throw an error.
```

Comment:
I think it'll be better if we treat the condition as normal downloading, without throwing exceptions.

Reply:
Ok.. would fix that.

Reply:
Hello @AlexV525 , just gave it a second thought. It might be better to throw an exception since the range might have been set in the header of the request. Having the range already set might lead to incomplete downloads.

It is written in the updated documentation that the range has to be specified in the header.

Reply:
> It is written in the updated documentation that the range has to be specified in the header.

Yes that's the point. Users may not required to pass the range header once they enable the resume. Length detection should be done inside the download, as well as the write mode detection. WDYT?

Reply:
Yeah.. The range header shouldn't be passed by Users. it should be done in the download.
That would be a better approach.

I would work on that.

Reply:
More thought on internally handling: If `download` automatically handles length, the flag might no longer be required.